quickwit/quickwit-ingest/src/ingest_v2/router.rs

```diff
@@ -396,22 +394,32 @@ impl EventSubscriber<LocalShardsUpdate> for Weak<RwLock<RouterState>> {
         let Some(state) = self.upgrade() else {
             return;
         };
-        // TODO: Insert the new shards in the shard table when `LocalShardsUpdate` also carries the
-        // leader ID.
+        let leader_id = local_shards_update.leader_id;
         let index_uid = local_shards_update.source_uid.index_uid;
         let source_id = local_shards_update.source_uid.source_id;
-        let closed_shard_ids: Vec<ShardId> = local_shards_update
-            .shard_infos
-            .into_iter()
-            .filter(|shard_info| shard_info.shard_state.is_closed())
-            .map(|shard_info| shard_info.shard_id)
-            .collect();
-
-        state
-            .write()
-            .await
-            .shard_table
+
+        let mut open_shard_ids: Vec<_> = Vec::new();
+        let mut closed_shard_ids: Vec<ShardId> = Vec::new();
+
+        for shard_info in local_shards_update.shard_infos {
+            match shard_info.shard_state {
+                ShardState::Open => open_shard_ids.push(shard_info.shard_id),
+                ShardState::Closed => closed_shard_ids.push(shard_info.shard_id),
+                _ => {}
```

Comment:
```suggestion
                _ => {}
```

nitpick, it is safer to put the other state than using `_` here.

Also, do we want to do something on Unavailable?